pkg/controller/migration/cni/calicotypes.go

```diff
@@ -0,0 +1,46 @@
+package cni
+
+// CalicoConf stores the common network config for Calico CNI plugin
+type CalicoConf struct {
+	CNIVersion string `json:"cniVersion,omitempty"`
+	Name       string `json:"name"`
+	Type       string `json:"type"`
+	IPAM       struct {
+		Name       string
+		Type       string   `json:"type"`
+		Subnet     string   `json:"subnet"`
+		AssignIpv4 *string  `json:"assign_ipv4"`
+		AssignIpv6 *string  `json:"assign_ipv6"`
+		IPv4Pools  []string `json:"ipv4_pools,omitempty"`
+		IPv6Pools  []string `json:"ipv6_pools,omitempty"`
+	} `json:"ipam,omitempty"`
+	// Args                 Args   `json:"args"`
+	MTU                  int    `json:"mtu"`
+	Nodename             string `json:"nodename"`
+	NodenameFileOptional bool   `json:"nodename_file_optional"`
+	DatastoreType        string `json:"datastore_type"`
+	EtcdEndpoints        string `json:"etcd_endpoints"`
+	EtcdDiscoverySrv     string `json:"etcd_discovery_srv"`
+	LogLevel             string `json:"log_level"`
+	// Policy               Policy `json:"policy"`
+	// Kubernetes           Kubernetes        `json:"kubernetes"`
```

Comment:
There's a few fields on this struct which are not currently used by our migration code. As such, it seemed unnecessary to copy in the type definitions. I've commented them out in this file, but I think I'll remove them entirely.

Reply:
I think that is ok even when the fields are in the CNI config because it is an extra step to make sure unknown fields cause an error with unmarshalling.